tests/test_entities_generic_handler.py

```diff
@@ -0,0 +1,2945 @@
+            RuntimeError, self._get_syncer, mocked_sg,"entities_generic_bad_fields_formatting_missing_jira_field_key"
+        )
+
+    def test_bad_settings_formatting_for_status_mapping(self, mocked_sg):
+        """Test all the use cases where the status field mappings setting is not correctly formatted."""
+        # "sg_field" key must be defined in the field_mapping dictionary
+        self.assertRaises(
+            RuntimeError, self._get_syncer, mocked_sg,"entities_generic_bad_status_formatting_missing_sg_field_key"
+        )
+        # "mapping" key must be defined in the field_mapping dictionary
+        self.assertRaises(
+            RuntimeError, self._get_syncer, mocked_sg,"entities_generic_bad_status_formatting_missing_mapping_key"
+        )
+
+    def test_project_in_entity_mapping(self, mocked_sg):
+        """Test that a FPTR project entity cannot be used in the entity mapping."""
+        self.assertRaises(RuntimeError, self._get_syncer, mocked_sg, "entities_generic_with_project")
+
+    def test_fptr_missing_fields_in_schema(self, mocked_sg):
+        """Test that the FPTR entities used in the entity mapping are correctly setup in FPTR"""
+
+        # SHOTGUN_JIRA_ID_FIELD field must have been created for the FPTR entity
+        self.assertRaises(
+            RuntimeError, self._get_syncer, mocked_sg, "entities_generic_sg_entity_with_missing_field_in_schema"
+        )
+        # FPTR field associated to Jira "assignee" field must be an entity/multi-entity field
+        self.assertRaises(ValueError, self._get_syncer, mocked_sg, "entities_generic_bad_assignee_field_type")
+        # FPTR field associated to Jira "assignee" field must be an entity/multi-entity field supporting HumanUser
+        self.assertRaises(ValueError, self._get_syncer, mocked_sg, "entities_generic_bad_assignee_field_entity_type")
+
+
+# Mock Flow Production Tracking with mockgun, this works only if the code uses shotgun_api3.Shotgun
+# and does not `from shotgun_api3 import Shotgun` and then `sg = Shotgun(...)`
+@mock.patch("shotgun_api3.Shotgun")
+class TestEntitiesGenericHandlerFPTRToJira(TestEntitiesGenericHandler):
+    """Test the sync from FPTR to Jira, covering for the different type of entities."""
+
+    # -------------------------------------------------------------------------------
+    # FPTR to Jira Sync - Entity Change Event (entity creation/update)
+    # -------------------------------------------------------------------------------
+    def test_fptr_to_jira_entity_not_supported(self, mocked_sg):
+        """If an entity type is not supported (aka not defined in the settings), the event will be rejected."""
+        syncer, bridge = self._get_syncer(mocked_sg, name=self.HANDLER_NAME)
+        self.assertFalse(
+            bridge.sync_in_jira(
+                self.HANDLER_NAME,
+                "Shot",
+                mock_shotgun.SG_SHOT["id"],
+                mock_shotgun.SG_SHOT_CHANGE_EVENT,
+            )
+        )
```

Comment:
Is it also worth checking that we log a message? Is that possible?
That way we can be more confident that the reason it has come back false is for the reason we think it is. Otherwise, the test might pass but for an entirely different reason that caused the sync to be false.